numba/cuda/cudadrv/driver.py

```diff
@@ -1232,18 +1488,39 @@ def _rebuild(cls, handle_ary, size, source_info, offset):
 
 
 class MemoryPointer(object):
-    """A memory pointer that owns the buffer with an optional finalizer.
-
-    When an instance is deleted, the finalizer will be called regardless
-    of the `.refct`.
-
-    An instance is created with `.refct=1`.  The buffer lifetime
-    is tied to the MemoryPointer instance's lifetime.  The finalizer is invoked
-    only if the MemoryPointer instance's lifetime ends.
+    """A memory pointer that owns a buffer, with an optional finalizer. Memory
+    pointers provide reference counting, and instances are initialized with a
+    reference count of 1.
+
+    The base ``MemoryPointer`` class does not use the
+    reference count for managing the buffer lifetime. Instead, the buffer
+    lifetime is tied to the memory pointer instance's lifetime:
+
+    - When the instance is deleted, the finalizer will be called.
+    - When the reference count drops to 0, no action is taken.
+
+    Subclasses of ``MemoryPointer`` may modify these semantics, for example to
+    tie the buffer lifetime to the reference count, so that the buffer is freed
+    when there are no more references.
+
+    :param context: The context in which the pointer was allocated.
+    :type context: Context
+    :param pointer: The address of the buffer.
+    :type pointer: ctypes.c_void_p
+    :param size: The size of the allocation in bytes.
+    :type size: int
+    :param owner: The owner is sometimes set by the internals of this class, or used for
+                  Numba's internal memory management. It should not be provided
+                  by an external user of the ``MemoryPointer`` class (e.g. from
+                  within an EMM Plugin); the default of `None` should always
+                  suffice.
+    :type owner: NoneType
+    :param finalizer: A function that is called when the buffer is to be freed.
+    :type finalizer: function
     """
     __cuda_memory__ = True
 
-    def __init__(self, context, pointer, size, finalizer=None, owner=None):
+    def __init__(self, context, pointer, size, owner=None, finalizer=None):
```

Comment:
It makes the order of arguments is now consistent across `MemoryPointer`, `AutoFreePointer`, `MappedMemory`, and `PinnedMemory` - since this is bringing a private API into the public it seemed like a good time to do this cleanup (rather than leaving an inconsistent API).

Reply:
I'm happy with this reasoning, and agree with proposed change.